### PR TITLE
Save unsent emails to disk to be resent later

### PR DESCRIPTION
**Pull request by acasajus**
- Added a config SAVE_UNSENT_DIR to specify where to keep the unsent messages
- Added a method to save the messages to disk if sending fails for some reason
- Added method to load them from disk

### CODE REVIEW

File: app/mail_sender.py [hunk 1]
```diff
@@ -23,22 +30,72 @@ class SendRequest:
     is_forward: bool = False
     ignore_smtp_errors: bool = False
 
+    def to_bytes(self) -> bytes:
+        if not config.SAVE_UNSENT_DIR:
+            return
+        if not os.path.isdir(config.SAVE_UNSENT_DIR):
```
Comment (nguyenkims) [inline app/mail_sender.py:36]: I wonder if  we can do this directory creation in `config` instead? Similarly to https://github.com/simple-login/app/blob/15810c2382cc81e6312e7a8484cc47af1a278a78/app/config.py#L226

**Review by nguyenkims — COMMENTED** [review]
Should we have a cronjob to resend emails that can't be sent?

**Comment by acasajus** [comment]
Yeah, we should add a cronjob. But Id' prefer to do it in another PR to keep this one smallish.

**Comment by acasajus** [comment]
Why the heck the big buttton is `close with comment`?

File: app/mail_sender.py (hunk 1 above)
Comment (acasajus) [reply app/mail_sender.py]: Done

File: app/mail_sender.py [hunk 2]
```diff
@@ -98,20 +151,30 @@ def _send_to_smtp(self, send_request: SendRequest, retries: int):
...
                 self._send_to_smtp(send_request, retries - 1)
             else:
-                if send_request.ignore_smtp_error:
-                    LOG.w("Ignore smtp error %s", e)
-                else:
-                    raise
+                if send_request.ignore_smtp_errors:
+                    LOG.w(f"Ignore smtp error {e}")
+                    return
+                LOG.e(
```
Comment (nguyenkims) [inline app/mail_sender.py:166]: I wonder if we can use LOG.exception here as it'll include all the stacktrace?

**Review by nguyenkims — APPROVED** [review]
LGTM! Just left a suggestion :)

File: app/mail_sender.py (hunk 2 above)
Comment (acasajus) [reply app/mail_sender.py:166]: Changed :+1:

**Pull request merged**